src/player.cpp

```diff
@@ -7004,12 +7004,50 @@ bool player::consume_med( item &target )
     return target.charges <= 0;
 }
 
+static bool query_consume_ownership( item &target, player &p )
+{
+    if( target.has_owner() && target.get_owner() != p.get_faction() ) {
+        if( p.get_value( "THIEF_MODE" ) == "THIEF_ASK" ) {
+            Pickup::query_thief();
+            if( p.get_value( "THIEF_MODE" ) == "THIEF_HONEST" ) {
+                if( p.get_value( "THIEF_MODE_KEEP" ) != "YES" ) {
+                    p.set_value( "THIEF_MODE", "THIEF_ASK" );
+                }
+                return false;
+            }
+        }
+        std::vector<npc *> witnesses;
+        for( npc &elem : g->all_npcs() ) {
+            if( rl_dist( elem.pos(), p.pos() ) < MAX_VIEW_DISTANCE && elem.sees( p.pos() ) ) {
+                witnesses.push_back( &elem );
+            }
+            for( npc *elem : witnesses ) {
```

Comment:
I assume this and the following should be outside of the loop over `g->all_npcs()`.

Reply:
Ah yes.